File: app/src/main/java/club/hutcwp/lifeutil/ui/home/top/PhotoFragment.java
package club.hutcwp.lifeutil.ui.home.top;


import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.content.ContextCompat;
import android.support.v4.view.ViewPager;

import java.util.List;

import club.hutcwp.lifeutil.R;
import club.hutcwp.lifeutil.adpter.ViewPagerAdapter;
import club.hutcwp.lifeutil.databinding.FragmentGirlBinding;
import club.hutcwp.lifeutil.entitys.PhotoCategory;
import club.hutcwp.lifeutil.ui.MainActivity;
import club.hutcwp.lifeutil.ui.base.BaseFragment;
import club.hutcwp.lifeutil.ui.home.sub.picture.PictureFragment;
import hut.cwp.mvp.BindPresenter;

@BindPresenter(presenter =  PhotoPresenter.class)
public class PhotoFragment extends BaseFragment<PhotoPresenter,IHome> implements IHome<PhotoCategory> {

    private FragmentGirlBinding binding;

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_girl;
    }


    @Override
    protected void initViews() {
        binding = (FragmentGirlBinding) getBinding();
        binding.toolbar.setTitle(getString(R.string.gank));
        ((MainActivity) getActivity()).initDrawer(binding.toolbar);
    }


    @Override
    protected void lazyFetchData() {
        getPresenter().getCategory();
    }

    /**
     * 初始化TabLayout
     */
    @Override
    public void initTabLayout(List<PhotoCategory> photoCategories) {
        setUpViewPager(binding.viewPager, photoCategories);
        binding.viewPager.setOffscreenPageLimit(binding.viewPager.getAdapter().getCount());
        binding.tabLayout.setSelectedTabIndicatorColor(ContextCompat.getColor(getActivity(), R.color.white));
        binding.tabLayout.setupWithViewPager(binding.viewPager);
        binding.tabLayout.setTabMode(TabLayout.GRAVITY_CENTER);
    }

    @Override
    public void setUpViewPager(ViewPager viewPager, List<PhotoCategory> categories) {
        ViewPagerAdapter adapter = new ViewPagerAdapter(getChildFragmentManager());
        for (PhotoCategory category : categories) {
            Fragment categoryFragment = new PictureFragment();
            Bundle data = new Bundle();
            data.putInt("type",category.getType());
            data.putString("url", category.getUrl());
            categoryFragment.setArguments(data);
            adapter.addFrag(categoryFragment, category.getName());
        }
        viewPager.setAdapter(adapter);
    }
}

File: app/src/main/java/club/hutcwp/lifeutil/ui/home/top/PhotoFragment.kt
package club.hutcwp.lifeutil.ui.home.top


import android.os.Bundle

import com.google.android.material.tabs.TabLayout

import androidx.core.content.ContextCompat
import androidx.viewpager.widget.ViewPager

import club.hutcwp.lifeutil.R
import club.hutcwp.lifeutil.adpter.ViewPagerAdapter
import club.hutcwp.lifeutil.databinding.FragmentGirlBinding
import club.hutcwp.lifeutil.entitys.PhotoCategory
import club.hutcwp.lifeutil.ui.MainActivity
import club.hutcwp.lifeutil.ui.base.BaseFragment
import club.hutcwp.lifeutil.ui.home.sub.picture.PictureFragment
import hut.cwp.mvp.BindPresenter

@BindPresenter(presenter = PhotoPresenter::class)
class PhotoFragment : BaseFragment<PhotoPresenter, PhotoFragment>(), IHome<PhotoCategory> {

    private var fragmentGirlBinding: FragmentGirlBinding? = null

    override fun getLayoutId(): Int {
        return R.layout.fragment_girl
    }


    override fun initViews() {
        fragmentGirlBinding = getBinding() as FragmentGirlBinding
        fragmentGirlBinding!!.toolbar.title = getString(R.string.gank)
        (activity as MainActivity).initDrawer(fragmentGirlBinding!!.toolbar)
    }


    override fun lazyFetchData() {
        presenter.getCategory()
    }

    /**
     * 初始化TabLayout
     */
    override fun initTabLayout(categories: List<PhotoCategory>) {
        setUpViewPager(fragmentGirlBinding!!.viewPager, categories)
        fragmentGirlBinding!!.viewPager.offscreenPageLimit = fragmentGirlBinding!!.viewPager.adapter!!.count
        fragmentGirlBinding!!.tabLayout.setSelectedTabIndicatorColor(ContextCompat.getColor(activity!!, R.color.white))
        fragmentGirlBinding!!.tabLayout.setupWithViewPager(fragmentGirlBinding!!.viewPager)
        fragmentGirlBinding!!.tabLayout.tabMode = TabLayout.MODE_SCROLLABLE
    }

    override fun setUpViewPager(viewPager: ViewPager, categories: List<PhotoCategory>) {
        val adapter = ViewPagerAdapter(childFragmentManager)
        for (category in categories) {
            val categoryFragment = PictureFragment()
            val data = Bundle()
            data.putInt("type", category.type)
            data.putString("url", category.url)
            categoryFragment.arguments = data
            adapter.addFrag(categoryFragment, category.name)
        }
        viewPager.adapter = adapter
    }
}
